### PR TITLE
Restructure Synchronizing Template Repositories

### PR DESCRIPTION
**Pull request by adamlazik1**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2232162

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.8/Katello 4.10
* [x] Foreman 3.7/Katello 4.9 (planned Satellite 6.14)
* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (Satellite 6.13)
* [x] Foreman 3.4/Katello 4.6 (EL8 only)
* [x] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* [x] Foreman 3.2/Katello 4.4 on EL7 & EL8
* [x] Foreman 3.1/Katello 4.3 on EL7 & EL8 (Satellite 6.11 EL7/8; orcharhino 6.3 on EL7/8)
* We do not accept PRs for Foreman older than 3.1.

### CODE REVIEW

File: guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc [hunk 1]
```diff
@@ -28,20 +28,24 @@ organizations:
 ----
 
 .Procedure
-. Create the directory where templates are stored and apply appropriate permissions and SELinux context:
+. In `/var/lib/foreman`, create a directory for storing templates:
 +
 [subs="+quotes"]
 ----
-# mkdir -p _/usr/share/templates_dir/_
-# chown foreman _/usr/share/templates_dir/_
```
Comment (ekohl) [inline guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc:36]: This is still needed: if you create it as  root then Foreman can't write to it.

File: guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc [hunk 2]
```diff
@@ -28,20 +28,24 @@ organizations:
...
-# chcon -t httpd_sys_rw_content_t _/usr/share/templates_dir/_ -R
+# mkdir /var/lib/foreman/_My_Templates_Dir_
```
Comment (ekohl) [inline guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc:35]: Is this even needed? I'd think the plugin creates this by itself (if it has permissions).

cc @theforeman/foreman_templates

File: guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc [hunk 3]
```diff
@@ -28,20 +28,24 @@ organizations:
...
 ----
-. Change the *Repo* setting on the *TemplateSync* tab to match the export directory `/usr/share/templates_dir/`.
++
+[NOTE]
+====
+You can place your templates to a custom directory outside `/var/lib/foreman`, but you have to set the `foreman_lib_t` SELinux label to that directory.
+====
```
Comment (maximiliankolb) [inline guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc:41]:
```suggestion
If you want to place your templates to a custom directory outside `/var/lib/foreman`, you have to set the `foreman_lib_t` SELinux label to that directory.
```

I am unsure if a "note" admonition is required here.

File: guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc [hunk 4]
```diff
@@ -28,20 +28,24 @@ organizations:
...
+
```
Comment (maximiliankolb) [inline guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc:42]:
```suggestion
```

File: guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc [hunk 5]
```diff
@@ -28,20 +28,24 @@ organizations:
...
+. Change the *Repo* setting on the *TemplateSync* tab to match the export directory `/var/lib/foreman/_My_Templates_Dir_/`.
 . Export templates from your {ProjectServer} to a local directory:
 +
 [options="nowrap", subs="verbatim,quotes,attributes"]
 ----
-$ curl -H "Accept:application/json" \
+# curl -H "Accept:application/json" \
 -H "Content-Type:application/json" \
 -u _login_:__password__ \
 -k https://_{foreman-example-com}/api/v2/templates/export \
```
Comment (ekohl) [inline guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc:51]: Just below this is `-X POST \` but I think that should be `-X POST`.

**Review by ekohl — CHANGES_REQUESTED** [review]
In the preview I still see SELinux instructions that are wrong: https://theforeman-foreman-documentation-preview-pr-2414.surge.sh/nightly/Managing_Hosts/index-foreman-el.html#Exporting_Templates_managing-hosts uses `httpd_sys_rw_content_t` which it can't read. I'd also recommend placing that in `/var/lib/foreman`, similar to this.

File: guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc (hunk 1 above)
Comment (adamlazik1) [reply guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc:36]: Thanks, restored.

File: guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc (hunk 5 above)
Comment (adamlazik1) [reply guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc:56]: Thanks, fixed.

**Comment by adamlazik1** [comment]
> In the preview I still see SELinux instructions that are wrong: https://theforeman-foreman-documentation-preview-pr-2414.surge.sh/nightly/Managing_Hosts/index-foreman-el.html#Exporting_Templates_managing-hosts uses httpd_sys_rw_content_t which it can't read. I'd also recommend placing that in /var/lib/foreman, similar to this.

Edited the instructions, please verify.

File: guides/common/modules/proc_exporting-templates.adoc [hunk 6]
```diff
@@ -15,24 +15,30 @@ The {ProjectWebUI} displays the status of the export.
...
 [options="nowrap", subs="+quotes,attributes"]
 ----
-$ git clone https://github.com/theforeman/community-templates __/custom/templates__
+$ git clone https://github.com/theforeman/community-templates /var/lib/foreman
 ----
-. Change the owner of your local directory to the `foreman` user, and change the SELinux context with the following commands:
++
+[NOTE]
+====
+You can clone your repository to a custom directory outside `/var/lib/foreman`, but you have to set the `foreman_lib_t` SELinux label to that directory.
```
Comment (evgeni) [inline guides/common/modules/proc_exporting-templates.adoc:27]: "… but you have to ensure the Foreman service can read its contents (check file permissions and SELinux label being `foreman_lib_t`)"

or something along the lines -- SELinux is not the only one who will prevent things from working,

File: guides/common/modules/proc_exporting-templates.adoc [hunk 7]
```diff
@@ -15,24 +15,30 @@ The {ProjectWebUI} displays the status of the export.
 The status is not persistent; if you leave the status page, you cannot return to it.
 
 .CLI procedure
-. Clone a local copy of your Git repository:
+. Clone a local copy of your Git repository to `/var/lib/foreman`:
 +
 [options="nowrap", subs="+quotes,attributes"]
 ----
-$ git clone https://github.com/theforeman/community-templates __/custom/templates__
+$ git clone https://github.com/theforeman/community-templates /var/lib/foreman
```
Comment (evgeni) [inline guides/common/modules/proc_exporting-templates.adoc:22]:
```suggestion
$ git clone https://github.com/theforeman/community-templates /var/lib/foreman/community-templates
```

File: guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc (hunk 2 above)
Comment (ofedoren) [reply guides/common/modules/proc_synchronizing-templates-with-a-local-directory-using-the-api.adoc:35]: It depends on what values you provide in the export form.

`Repo` can be either full path or a "root" path. In this case it can be both `/var/lib/foreman/_My_Templates_Dir_` and `/var/lib/foreman`. In this case if `_My_Templates_Dir` doesn't exist, it won't be created.
`Dirname` can be omitted or specified. If specified then the plugin will create this directory in `Repo` root.

TL;DR:
`_My_Templates_Dir_` won't be created unless it's specified as `Dirname` in export form.

File: guides/common/modules/proc_exporting-templates.adoc [hunk 8]
```diff
@@ -15,24 +15,30 @@ The {ProjectWebUI} displays the status of the export.
 The status is not persistent; if you leave the status page, you cannot return to it.
 
 .CLI procedure
-. Clone a local copy of your Git repository:
+. Clone a local copy of your Git repository to `/var/lib/foreman`:
```
[the remark this answers is not in the document]
Comment (adamlazik1) [reply guides/common/modules/proc_exporting-templates.adoc]: Thank you, applied.

File: guides/common/modules/proc_exporting-templates.adoc (hunk 6 above)
Comment (adamlazik1) [reply guides/common/modules/proc_exporting-templates.adoc]: Thank you, applied.

File: guides/common/modules/proc_exporting-templates.adoc (hunk 7 above)
Comment (adamlazik1) [reply guides/common/modules/proc_exporting-templates.adoc]: Applied.

**Comment by adamlazik1** [comment]
Thank you. Applied feedback.

File: guides/common/modules/proc_exporting-templates.adoc [hunk 9]
```diff
@@ -15,24 +15,31 @@ The {ProjectWebUI} displays the status of the export.
 The status is not persistent; if you leave the status page, you cannot return to it.
 
 .CLI procedure
-. Clone a local copy of your Git repository:
+. Clone a local copy of your Git repository to a subdirectory of `/var/lib/foreman`:
```
Comment (ekohl) [inline guides/common/modules/proc_exporting-templates.adoc:18]: Why do we clone something? The templates plugin can also sync by itself and push.

https://github.com/theforeman/foreman_templates/blob/01d2dca4dd1deb7982cc6cd3a8f71e100f1b5282/app/services/foreman_templates/template_exporter.rb#L30-L61

So if you just use a git URL in the settings it should clone it, make the changes and push it back. That would remove all the need for directory permissions etc.

**Review by ekohl — COMMENTED** [review]
Sorry for escalating this, but I think it would be better to avoid the whole local files thing. The templates plugin can also sync by itself and push:

https://github.com/theforeman/foreman_templates/blob/01d2dca4dd1deb7982cc6cd3a8f71e100f1b5282/app/services/foreman_templates/template_exporter.rb#L30-L61

So if you just use a git URL in the settings it should clone it, make the changes and push it back. That would remove all the need for directory permissions etc.

Almost everyone has some internal git server, like GitHub Enterprise or GitLab. Local files is the least interesting option.

There are some instructions about SSH hidden under "Synchronizing Templates Using the Foreman API". That makes me think the structure is also poor. Reading it, I think you want that under the `Configuring` section.

From there on you have the problem that @asteflova is investigating: how do you have a procedure that can be executed via multiple implementations: Hammer & API (and to be written: Ansible via [theforeman.foreman.templates_import](https://theforeman.github.io/foreman-ansible-modules/develop/plugins/templates_import_module.html#ansible-collections-theforeman-foreman-templates-import-module)).

**Comment by adamlazik1** [comment]
Temporarily stopping work on this due to higher priority tickets.

File: guides/common/modules/proc_exporting-templates.adoc (hunk 9 above)
Comment (adamlazik1) [reply guides/common/modules/proc_exporting-templates.adoc:18]: @ekohl what would be exact steps here? Where is the setting to which I need to put the URL and what do I need to do after that to achieve the same result as it is at the end of this procedure?

**Comment by ekohl** [comment]
Looking at the structure today we have this:
```
13. Synchronizing Template Repositories
13.1. Enabling the TemplateSync Plug-in
13.2. Configuring the TemplateSync Plug-in
13.3. Importing and Exporting Templates
13.3.1. Importing Templates
13.3.2. Exporting Templates
13.3.3. Synchronizing Templates Using the Foreman API
13.3.4. Synchronizing Templates with a Local Directory Using the Foreman API
13.4. Advanced Git Configuration
13.5. Uninstalling the Foreman Templates Plug-in
```

The first thing I'd change is move things around.

Under 13.3.3 there is code to generate an SSH keypair, and accepting the SSH key. IMHO that's a separate procedure that's closer to be part of configuring the plugin. You need it for both important and exporting when the remote is SSH, so also 13.3.1. Perhaps SSH config should be 13.2.1?

Then we have 13.3.2 where it's exported. You can change that to Hammer command to use `--repo git@/github.com:theforeman/community-templates` and skip steps 1 & 2 (to clone and fix permissions). You can add a note that it clones the repository, applies the changes in a commit and pushes the changes. You can point out that `--branch MyBranch` can be used. That way we avoid the local directory.

For context: the reason a local repository is (probably) used (I'm not the author) is that pushing to the remote repository doesn't allow review. I'd suggest that a better way is to first create a new branch, then use `--branch` to make foreman_templates work isolated. Then you can submit a PR and do your review.

Then step 13.3.4 can remain, but I wonder if you need to do the API parts. Perhaps it should be `13.4 Synchronizing Templates with a Local Directory` to set up the local directory and then state you sync using the above chapters, either with Hammer or the API.

Changing it up again, perhaps it should be this structure:

```
13. Synchronizing Template Repositories
13.1. Enabling the TemplateSync Plug-in
13.2. Configuring the TemplateSync Plug-in
13.3. Using Repository Sources
13.3.1. HTTPS
13.3.1.1. Accepting a self-signed certificate
13.3.2. SSH
13.3.3. Local directory
13.4. Importing and Exporting Templates
13.4.1. Importing Templates
13.4.1.1. UI
13.4.1.2. Hammer
13.4.1.3. Foreman API
13.4.2. Exporting Templates
13.4.2.1. UI
13.4.2.2. Hammer
13.4.2.3. Foreman API
13.5. Uninstalling the Foreman Templates Plug-in
```

Does that make sense? Just looking ahead: for importing and exporting we can also add Ansible at some point.

File: guides/common/modules/proc_exporting-templates.adoc [hunk 10]
```diff
@@ -15,24 +15,31 @@ The {ProjectWebUI} displays the status of the export.
...
-# chcon -R -t httpd_sys_rw_content_t __/custom/templates__
+# chown -R foreman:foreman /var/lib/foreman/community-templates
 ----
 . To export the templates to your local repository, enter the following command:
 +
 [options="nowrap", subs="+quotes,attributes"]
 ----
-hammer export-templates --organization 'Default Organization' --repo __/custom/templates__
+hammer export-templates \
+--organization 'Default Organization' \
```
Comment (mjivraja) [inline guides/common/modules/proc_exporting-templates.adoc:41]:
```suggestion
--organization "Default Organization" \
```
In some places, we use double quotes. Not sure which one to follow.

File: guides/common/modules/proc_exporting-templates.adoc [hunk 11]
```diff
@@ -15,24 +15,31 @@ The {ProjectWebUI} displays the status of the export.
...
 [options="nowrap", subs="+quotes,attributes"]
 ----
-$ git clone https://github.com/theforeman/community-templates __/custom/templates__
+$ git clone https://github.com/theforeman/community-templates /var/lib/foreman/community-templates
 ----
-. Change the owner of your local directory to the `foreman` user, and change the SELinux context with the following commands:
++
+[NOTE]
+====
+You can clone your repository to a custom directory outside `/var/lib/foreman`, but you have to ensure that the `Foreman` service can read its contents.
```
Comment (mjivraja) [inline guides/common/modules/proc_exporting-templates.adoc:27]:
```suggestion
You can clone your repository to a custom directory outside `/var/lib/foreman`.
Ensure that the `Foreman` service can read its contents.
```
Minimalism.

File: guides/common/modules/proc_exporting-templates.adoc (hunk 10 above)
Comment (adamlazik1) [reply guides/common/modules/proc_exporting-templates.adoc]: Fixed.

File: guides/common/modules/proc_exporting-templates.adoc (hunk 11 above)
Comment (adamlazik1) [reply guides/common/modules/proc_exporting-templates.adoc]: I'd prefer to keep this as one sentence as it is a direct warning that is the user wishes  to clone to a custom directory, they must ensure that these conditions are met.

File: guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+= Synchronizing Templates with an Existing Repository
+
+You can use existing repositories to import and export templates from your {ProjectServer}.
+
+.Procedure
+. Configure a version control system that uses SSH authorization, for example gitosis, gitolite, or git daemon.
+. Configure the TemplateSync plug-in settings on a *TemplateSync* tab.
+.. Change the *Branch* setting to match the target branch on a Git server.
+.. Change the *Repo* setting to match the Git repository.
+For example, for the repository located in `git@git.example.com/templates.git` set the setting into `ssh://git@git.example.com/templates.git`.
```
Comment (ekohl) [inline guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc:11]: I'm not sure `ssh://` is really needed. AFAIK the ruby-git library that we use doesn't care and handles URLs like the regular git client.

File: guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+. Accept Git SSH host key as the `foreman` user:
++
+[subs="+quotes"]
+----
+# sudo -u foreman ssh _git.example.com_
+----
++
+You can see the `Permission denied, please try again.` message in the output, which is expected, because the SSH connection cannot succeed yet.
```
Comment (ekohl) [inline guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc:19]: I would swap the order: create a key, instruct the user to set up the remote end and then use SSH to perform an end-to-end test.

File: guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+
+You can use existing repositories to import and export templates from your {ProjectServer}.
+
+.Procedure
+. Configure a version control system that uses SSH authorization, for example gitosis, gitolite, or git daemon.
+. Configure the TemplateSync plug-in settings on a *TemplateSync* tab.
+.. Change the *Branch* setting to match the target branch on a Git server.
+.. Change the *Repo* setting to match the Git repository.
+For example, for the repository located in `git@git.example.com/templates.git` set the setting into `ssh://git@git.example.com/templates.git`.
+. Accept Git SSH host key as the `foreman` user:
```
Comment (ekohl) [inline guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc:12]: When using SSH, accept the SSH key. Because if a user only uses HTTPS then it's not needed.

File: guides/common/modules/proc_accepting-a-self-signed-git-certificate.adoc [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+= Accepting a Self-signed Git Certificate
+
+If you are using a self-signed certificate authentication on your Git server, you must validate the certificate.
+
+.Procedure
+* Validate the certificate:
++
+[options="nowrap" subs="+quotes,verbatim,attributes"]
+----
+# sudo -u foreman git config --global http.sslCAPath _Path_To_My_Certificate_
```
Comment (ekohl) [inline guides/common/modules/proc_accepting-a-self-signed-git-certificate.adoc:11]: I'd say this step is roughly the HTTPS equivalent of accepting the SSH host key, which you have `in 13.2.1. Synchronizing Templates with an Existing Repository`.

Would it make sense to have 2 sub-procedures in `13.2.1. Synchronizing Templates with an Existing Repository`. One to set up SSH, one to set up HTTPS?

File: guides/common/modules/proc_exporting-templates.adoc [hunk 16]
```diff
@@ -3,6 +3,9 @@
 
 You can export templates to a version control server, such as a Git repository.
```
Comment (ekohl) [inline guides/common/modules/proc_exporting-templates.adoc:4]: I know it was already there before, but it only supports git as a VCS. This text could imply another VCS (such as mercurial or subversion) is supported.

**Review by ekohl — CHANGES_REQUESTED** [review]
> @ekohl I did most of the changes you requested but I am unable to find more info on the HTTPS procedure of Using Repository Sources. Could you elaborate on that? The rest is ready for review.

I think I covered it: there's the `git config` line that should be equivalent to the SSH setup part.

File: guides/common/modules/proc_exporting-templates.adoc (hunk 11 above)
Comment (mjivraja) [reply guides/common/modules/proc_exporting-templates.adoc]: You are already on `Note` so the importance remains the same as you want. But, final call is yours :+1:

File: guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc (hunk 12 above)
Comment (adamlazik1) [reply guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc]: Removed.

File: guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc (hunk 13 above)
Comment (adamlazik1) [reply guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc]: Swapped.

File: guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc (hunk 14 above)
Comment (adamlazik1) [reply guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc]: Fixed.

File: guides/common/modules/proc_synchronizing-templates-with-a-local-directory.adoc [hunk 17]
```diff
@@ -0,0 +1,49 @@
+[id="Synchronizing_Templates_with_a_Local_Directory_{context}"]
```
Comment (mjivraja) [inline guides/common/modules/proc_synchronizing-templates-with-a-local-directory.adoc:1]:
```suggestion
[id="Synchronizing_Templates_with_a_Local_Directory_{context}"]
```
Choose from either Sentence case or lower case.

File: guides/common/modules/proc_accepting-a-self-signed-git-certificate.adoc (hunk 15 above)
Comment (adamlazik1) [reply guides/common/modules/proc_accepting-a-self-signed-git-certificate.adoc]: Instead of splitting into two procedures, since there would be some common steps, I added validating the certificate as the last step. Please verify.

File: guides/common/modules/proc_exporting-templates.adoc (hunk 16 above)
Comment (adamlazik1) [reply guides/common/modules/proc_exporting-templates.adoc]: Fixed.

File: guides/common/modules/proc_synchronizing-templates-with-a-local-directory.adoc (hunk 17 above)
Comment (adamlazik1) [reply guides/common/modules/proc_synchronizing-templates-with-a-local-directory.adoc:1]: Actually, Camel Case is very common. You can verify with `git grep -E 'id="[A-Z]'`

Comment (mjivraja) [reply guides/common/modules/proc_synchronizing-templates-with-a-local-directory.adoc:1]: Right, but here `proc_synchronizing-templates-with-an-existing-repository.adoc` you have used the smallcase. It should be the same for all?

Or, is it flexible?

Comment (adamlazik1) [reply guides/common/modules/proc_synchronizing-templates-with-a-local-directory.adoc:1]: I believe it is flexible. Smallcase is generated by `newdoc` while some people who create modules manually use Camel Case. So far both styles are accepted.

Comment (mjivraja) [reply guides/common/modules/proc_synchronizing-templates-with-a-local-directory.adoc:1]: :+1:

File: guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc
Comment (ekohl) [inline guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc:1]: Is it logical to have steps as follows:

1. HTTPS
2. SSH
    1. Create SSH key
    2. Configure remote server
    3. Accept SSH host key
3. Configure repo
    1. Change branch
    2. Change repo

Comment (adamlazik1) [reply guides/common/modules/proc_synchronizing-templates-with-an-existing-repository.adoc:1]: Fixed.

**Review by ekohl — APPROVED** [review]
I didn't read the final fully generated version, but by reading the individual changes and previously generated files I'd say that this is ready from a tech review perspective. Given the non-trivial size I've added a label that it needs a style review.

**Comment by adamlazik1** [comment]
Thanks!

File: guides/common/modules/proc_exporting-templates.adoc [hunk 18]
```diff
@@ -14,25 +17,51 @@ For more information about each field, see xref:Configuring_the_TemplateSync_plu
...
+. Send a `POST` request to `api/v2/templates/export`:
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
 ----
-# chown -R foreman:foreman __/custom/templates__
-# chcon -R -t httpd_sys_rw_content_t __/custom/templates__
+# curl -H "Accept:application/json" \
+-H "Content-Type:application/json" \
+-u _login_:__password__ \
+-k https://_{foreman-example-com}/api/v2/templates/export \
```
Comment (mjivraja) [inline guides/common/modules/proc_exporting-templates.adoc:47]:
```suggestion
-k https://_{foreman-example-com}_/api/v2/templates/export \
```

File: guides/common/modules/proc_exporting-templates.adoc [hunk 19]
```diff
@@ -14,25 +17,51 @@ For more information about each field, see xref:Configuring_the_TemplateSync_plu
...
+[id="cli_Exporting_Templates_{context}"]
 .CLI procedure
-. Clone a local copy of your Git repository:
+. To export the templates to a repository, enter the following command:
 +
 [options="nowrap", subs="+quotes,attributes"]
 ----
-$ git clone https://github.com/theforeman/community-templates __/custom/templates__
+hammer export-templates \
+--organization "Default Organization" \
```
Comment (mjivraja) [inline guides/common/modules/proc_exporting-templates.adoc:27]:
```suggestion
--organization "_My Organization_" \
```
Could this be better? Or `_Default Organization_`

File: guides/common/modules/proc_exporting-templates.adoc [hunk 20]
```diff
@@ -14,25 +17,51 @@ For more information about each field, see xref:Configuring_the_TemplateSync_plu
...
 ----
-. To export the templates to your local repository, enter the following command:
 +
-[options="nowrap", subs="+quotes,attributes"]
+If the export is successful, you receive `{"message":"Success"}`.
+
+[NOTE]
+====
+You can override default API settings by specifying them in the request with the `-d` parameter.
+The following example exports templates to the `git.example.com/templates` repository:
```
Comment (mjivraja) [inline guides/common/modules/proc_exporting-templates.adoc:56]:
```suggestion
The following example exports templates to the `_git.example.com_/templates` repository:
```

File: guides/common/modules/proc_exporting-templates.adoc [hunk 21]
```diff
@@ -14,25 +17,51 @@ For more information about each field, see xref:Configuring_the_TemplateSync_plu
...
+
+[options="nowrap", subs="verbatim,quotes,attributes"]
 ----
-hammer export-templates --organization 'Default Organization' --repo __/custom/templates__
+# curl -H "Accept:application/json" \
+-H "Content-Type:application/json" \
+-u login:password \
+-k https://{foreman-example-com}/api/v2/templates/export \
```
Comment (mjivraja) [inline guides/common/modules/proc_exporting-templates.adoc:63]:
```suggestion
-k https://_{foreman-example-com}_/api/v2/templates/export \
```

File: guides/common/modules/proc_importing-templates.adoc [hunk 22]
```diff
@@ -53,3 +63,19 @@ $ hammer import-templates \
...
+.API Procedure
+
+. Send a `POST` request to `api/v2/templates/import`:
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# curl -H "Accept:application/json" \
+-H "Content-Type:application/json" \
+-u _login_:__password__ \
+-k https://_{foreman-example-com}/api/v2/templates/import \
```
Comment (mjivraja) [inline guides/common/modules/proc_importing-templates.adoc:77]:
```suggestion
-k https://_{foreman-example-com}_/api/v2/templates/import \
```

File: guides/common/modules/proc_exporting-templates.adoc (hunk 18 above)
Comment (adamlazik1) [reply guides/common/modules/proc_exporting-templates.adoc]: Fixed.

File: guides/common/modules/proc_exporting-templates.adoc (hunk 20 above)
Comment (adamlazik1) [reply guides/common/modules/proc_exporting-templates.adoc:56]: The whole code block is presented as an example so in this case I will leave it as is.

File: guides/common/modules/proc_exporting-templates.adoc (hunk 21 above)
Comment (adamlazik1) [reply guides/common/modules/proc_exporting-templates.adoc]: Applied.

File: guides/common/modules/proc_importing-templates.adoc (hunk 22 above)
Comment (adamlazik1) [reply guides/common/modules/proc_importing-templates.adoc]: Applied.

**Review by mjivraja — APPROVED** [review]
LGTM!

**Comment by adamlazik1** [comment]
Thank you for review @mjivraja.

**Comment by adamlazik1** [comment]
Merged and cherry-picked:
   6ff936b25..456a99a06  3.1 -> 3.1
   958f9d0fb..b0eda226f  3.2 -> 3.2
   f67902d68..6ac675985  3.3 -> 3.3
   75dfa80de..e575e2e1f  3.4 -> 3.4
   54de7d58a..672bf3cb9  3.5 -> 3.5
   1cc013913..0f04dcb86  3.6 -> 3.6
   c58546a26..cd1ab293e  3.7 -> 3.7
   c4aec1828..39ce2ebc6  3.8 -> 3.8
